Amazon.Ion.ObjectMapper/IonSerializer.cs

```diff
@@ -127,6 +128,8 @@ public class IonSerializationOptions
         public string[] AnnotatedTypeAssemblies { get; init; } = new string[] {};
 
         public readonly bool PermissiveMode;
+        
+        public Dictionary<Type, dynamic> IonSerializers { get; init; }

```

Comment:
I think this can be slightly improved by using `Dictionary<Type, IonSerializer<dynamic>>`.

Reply:
That at least says the value must be an `IonSerializer`

Reply:
That doesn't work because `IonSerializer<dynamic>` is understood as a serializer for dynamic objects, not a serializer for any type.
ie. You can't assign a `IonSerializer<string>` to `IonSerializer<dynamic>`.